libraries/AP_Compass/AP_Compass_AK8963.cpp

```diff
@@ -189,18 +167,19 @@ bool AP_Compass_AK8963::init()
 
     /* register the compass instance in the frontend */
     _compass_instance = register_compass();
-    set_dev_id(_compass_instance, AP_COMPASS_TYPE_AK8963_MPU9250);
+
+    set_dev_id(_compass_instance, _bus->get_dev_id());
```

Comment:
It is indeed weird. The reason for that is that set_dev_id is a member of the AP_Compass_Backend class and so, the bus would need to access a method from the backend. Which is, I think even worse. Do you see a way that in a clean fashion?